Source/DataSources/DataSource.js

```diff
@@ -58,15 +58,15 @@ Object.defineProperties(DataSource.prototype, {
   /**
    * Gets an event that will be raised if an error is encountered during processing.
    * @memberof DataSource.prototype
-   * @type {Event}
+   * @type {Event<function(this, RequestErrorEvent)>}
    */
   errorEvent: {
     get: DeveloperError.throwInstantiationError,
   },
   /**
    * Gets an event that will be raised when the value of isLoading changes.
    * @memberof DataSource.prototype
-   * @type {Event}
+   * @type {Event<function(this, boolean)>}
```

Comment:
I was worried about how JSDoc would handle this, but I _think_ this is okay, assuming we can get Event as a template working a little better.:

![image](https://user-images.githubusercontent.com/1495196/83275476-30453200-a19d-11ea-8f98-cf1cd2cac42d.png)

However, I'm pretty sure this breaks duct typing because CustomDataSource/CzmlDataSource/KmlDataSource/GeoJsonDataSource don't have the same signatures.

I'm not sure if the best course of action is to update the signatures, or pull out the Event related changes in a separate PR for more discussion.  I'll finish up the rest of the review and think about it.

Reply:
FWIW I have lines like

````ts
let src: DataSource;
...
src = new KmlDataSource(...)
````

in my code and it didn't break when compiling against this branch's typings.  I can see where you'd be concerned, though, so I understand if you want to hold off.  (If there's no rush, I could always fix all uses of Event on Monday, it's not actually that many.)